Makefile

```diff
@@ -73,13 +73,14 @@ run: build-local-container kind-load deploy ## Build image and run operator in-c
 
 # Binary builds
 GO_BUILD := $(Q)go build
+GOOS_TAG=$(if $(GOOS),-$(GOOS))
```

Comment:
Some folks might intentionally set GOOS in their local environment, which would result in this suffix always being added.

I wonder if we could do something like:
```
build-local-container: export BIN_SUFFIX=-$(GOOS)
```

And then in the go build lines:
```
CGO_ENABLED=0 go build -o $@$(BIN_SUFFIX) ./provisioner/plain-v0
```